src/ui/src/builder/BuilderTemplateInput.vue

```diff
@@ -0,0 +1,226 @@
+<template>
+	<div class="BuilderTemplateInput">
+		<template v-if="!props.multiline">
+			<input
+				ref="input"
+				class="templateInput"
+				type="text"
+				:value="props.value"
+				autocorrect="off"
+				autocomplete="off"
+				spellcheck="false"
+				:placeholder="props.placeholder"
+				:list="props.options ? `list-${props.inputId}` : undefined"
+				@input="handleInput"
+				@blur="handleBlur"
+			/>
+			<datalist v-if="props.options" :id="`list-${props.inputId}`">
+				<option
+					v-for="(option, optionKey) in options"
+					:key="optionKey"
+					:value="optionKey"
+				>
+					<template
+						v-if="option.toLowerCase() !== optionKey.toLowerCase()"
+					>
+						{{ option }}
+					</template>
+				</option>
+			</datalist>
+		</template>
+
+		<template v-if="props.multiline">
+			<textarea
+				ref="input"
+				v-capture-tabs
+				class="templateInput"
+				variant="code"
+				:value="props.value"
+				autocorrect="off"
+				autocomplete="off"
+				spellcheck="false"
+				:placeholder="props.placeholder"
+				@input="handleInput"
+			></textarea>
+		</template>
+
+		<div v-if="autocompleteOptions.length" class="fieldStateAutocomplete">
+			<div
+				v-for="(option, optionKey) in autocompleteOptions"
+				:key="optionKey"
+				class="fieldStateAutocompleteOption"
+				:value="optionKey"
+				@click="() => handleComplete(option.text)"
+			>
+				<span class="prop">{{ option.text }}</span
+				><span class="type">{{ option.type }}</span>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { inject, ref, nextTick } from "vue";
+import injectionKeys from "../injectionKeys";
+import Fuse from "fuse.js";
```

Comment:
I agree on avoiding library just for fuzzy research. I would recommand to write a utility function to perform fuzzy research if we need it. The algorithme looks simple.

Reply:
You shouldn't look at package size this way... 
Here you have real size of this package in production build:
<img width="761" alt="Screenshot 2024-06-11 at 14 29 34" src="https://github.com/writer/writer-framework/assets/649972/ebd0747c-9ab4-43db-88a1-cb11cb92b471">
But if you insist it could be minified further to something like 11.1kB (4.2 kB gziped)

Reply:
Wonder why it didn't use minified version...
<img width="856" alt="Screenshot 2024-06-11 at 14 38 39" src="https://github.com/writer/writer-framework/assets/649972/aa554020-0418-4b07-bb44-4661dcaffc34">

Reply:
maybe vite is minimizing it by itself? 🤔